diff --git a/scripts/configuration.sh b/scripts/configuration.sh
index 649b73c..3f54a52 100644
--- a/scripts/configuration.sh
+++ b/scripts/configuration.sh
@@ -52,3 +52,7 @@ function rbx_write_revision_file {
 function rbx_release_name {
   echo "rubinius-$(rbx_revision_version).tar.bz2"
 }
+
+function rbx_heroku_release_name {
+  echo "ruby-2.2.2-rbx-$(rbx_revision_version).tar.bz2"
+}
diff --git a/scripts/deploy.sh b/scripts/deploy.sh
index cae4b6f..e277253 100755
--- a/scripts/deploy.sh
+++ b/scripts/deploy.sh
@@ -92,6 +92,35 @@ function rbx_deploy_homebrew_binary {
   fi
 }
 
+# Build and upload a Heroku binary to S3.
+function rbx_deploy_heroku_binary {
+  if [[ $1 != linux ]]; then
+    return
+  fi
+
+  if [[ $(rbx_dist_version) != "14.04" ]]; then
+    return
+  fi
+
+  echo "Deploying Heroku binary $(rbx_heroku_release_name)..."
+
+  "$__dir__/package.sh" heroku || fail "unable to build Heroku binary"
+
+  local -a file_exts=("" ".sha1")
+  local url bucket name
+
+  bucket="$(rbx_binary_bucket)"
+  url="$(rbx_url_prefix "$bucket")"
+  name="$(rbx_heroku_release_name)"
+
+  for ext in "${file_exts[@]}"; do
+    if [[ -f $src$ext ]]; then
+      rbx_s3_upload "$url" "$bucket" "$name$ext" "$name$ext" "/ubuntu/cedar-14/" ||
+        fail "unable to upload file"
+    fi
+  done
+}
+
 # Build and upload a binary to S3.
 function rbx_deploy_travis_binary {
   local os_name vendor_llvm
@@ -346,6 +375,9 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
       "travis")
         rbx_deploy_travis_binary "$TRAVIS_OS_NAME"
         ;;
+      "heroku")
+        rbx_deploy_heroku_binary "$TRAVIS_OS_NAME"
+        ;;
       "travis-12.04")
         rbx_deploy_travis_12_04_binary
         ;;
diff --git a/scripts/package.sh b/scripts/package.sh
index 354def3..3675189 100755
--- a/scripts/package.sh
+++ b/scripts/package.sh
@@ -33,6 +33,14 @@ function rbx_package_homebrew {
   rbx_digest_file "$(rbx_release_name)" "sha1"
 }
 
+function rbx_package_heroku {
+  echo "Packaging for Heroku..."
+
+  rake package:heroku
+
+  rbx_digest_file "$(rbx_heroku_release_name)" "sha1"
+}
+
 if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
   case "$1" in
     "tar")
@@ -44,6 +52,9 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
     "homebrew")
       rbx_package_homebrew
       ;;
+    "heroku")
+      rbx_package_heroku
+      ;;
     *)
       echo "Usage: ${0##*/} package_type"
       exit 1
